tests/Unit/Evolution/VariableFixing/Test_LimitLorentzFactor.cpp

```diff
@@ -0,0 +1,85 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/DotProduct.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/VariableFixing/LimitLorentzFactor.hpp"
+#include "PointwiseFunctions/Hydro/LorentzFactor.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+// IWYU pragma: no_forward_declare Tensor
+
+namespace VariableFixing {
+namespace {
+void test_variable_fixer(const LimitLorentzFactor& variable_fixer) {
+  const Scalar<DataVector> density{DataVector{1., 2., 1.e-5, 1e-6}};
+  auto spatial_metric =
+      make_with_value<tnsr::ii<DataVector, 3, Frame::Inertial>>(density, 1.0);
+  get<0, 0>(spatial_metric) = 0.2;
+  get<0, 1>(spatial_metric) = 0.3;
+  get<0, 2>(spatial_metric) = 0.4;
+  get<1, 1>(spatial_metric) = 0.5;
+  get<1, 2>(spatial_metric) = 0.6;
+  get<2, 2>(spatial_metric) = 0.7;
+
+  const double value = 4.999e-1;
+  tnsr::I<DataVector, 3, Frame::Inertial> spatial_velocity{
+      {{DataVector{0.4999, 0.4999, 0.4999, 0.4999},
+        DataVector{1.3567, 1.3566, 1.3567, 1.3566},
+        DataVector{-0.200, -0.200, -0.200, -0.200}}}};
+
+  auto lorentz_factor = hydro::lorentz_factor(
+      dot_product(spatial_velocity, spatial_velocity, spatial_metric));
+  const Scalar<DataVector> expected_lorentz_factor{
+      DataVector{get(lorentz_factor)[0], get(lorentz_factor)[1], 50.,
+                 get(lorentz_factor)[3]}};
+  const double rescale_velocity_factor =
+      sqrt((1. - 1. / square(get(expected_lorentz_factor)[2])) /
+           (1. - 1. / square(get(lorentz_factor)[2])));
+  const tnsr::I<DataVector, 3, Frame::Inertial> expected_spatial_velocity{
+      {{DataVector{0.4999, 0.4999, 0.4999 * rescale_velocity_factor, 0.4999},
+        DataVector{1.3567, 1.3566, 1.3567 * rescale_velocity_factor, 1.3566},
+        DataVector{-0.200, -0.200, -0.200 * rescale_velocity_factor, -0.200}}}};
+
+  variable_fixer(make_not_null(&spatial_velocity),
+                 make_not_null(&lorentz_factor), density);
+
+  CHECK(lorentz_factor == expected_lorentz_factor);
```

Comment:
Please check that `expected_lorentz_factor == hydro::lorentz_factor(dot_product(expected_spatial_velocity...`